components/autocomplete/Autocomplete.js

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import Awesomplete from 'awesomplete';
+import SelectedItem from './SelectedItem';
+import './Autocomplete.scss';
+import { noop } from 'proton-shared/lib/helpers/function';
+
+const Autocomplete = ({
+    list,
+    selectedItems,
+    inputValue,
+    onRemove,
+    onSubmit,
+    onSelect,
+    onInputValueChange,
+    onOpen,
+    onClose,
+    onHighlight,
+    ...rest
+}) => {
+    let [awesomplete, setAwesomplete] = useState(null);
+    const inputRef = useRef(null);
+    const containerRef = useRef(null);
+
+    const handleInputValueChange = ({ target }) => {
+        onInputValueChange(target.value);
+    };
+
+    const handleSubmit = (item = { label: inputValue, value: inputValue, invalid: true }) => {
+        if (item.value) {
+            onSubmit(item);
+        }
+    };
+
+    const handleSelect = ({ text }) => {
+        handleSubmit(text);
+        onSelect(text);
+    };
+
+    const handleKeyDown = (e) => {
+        if (e.key === 'Backspace' && !inputValue) {
+            onRemove(selectedItems.length - 1);
+        }
+    };
+
+    useEffect(() => {
+        const awesompleteInstance = new Awesomplete(inputRef.current, {
+            container: () => containerRef.current,
+            ...rest
+        });
+
+        setAwesomplete(awesompleteInstance);
+
+        inputRef.current.addEventListener('awesomplete-selectcomplete', handleSelect);
+        inputRef.current.addEventListener('awesomplete-close', onClose);
+        inputRef.current.addEventListener('awesomplete-highlight', onHighlight);
+        inputRef.current.addEventListener('awesomplete-open', onOpen);
+
+        return () => {
+            awesompleteInstance.destroy();
+            inputRef.current.removeEventListener('awesomplete-selectcomplete', handleSelect);
+            inputRef.current.removeEventListener('awesomplete-close', onClose);
+            inputRef.current.removeEventListener('awesomplete-highlight', onHighlight);
+            inputRef.current.removeEventListener('awesomplete-open', onOpen);
+        };
+    }, []);
+
+    useEffect(() => {
+        if (awesomplete) {
+            awesomplete.list = list;
+            awesomplete.evaluate();
+        }
+    }, [awesomplete, list]);
+
+    return (
+        <form
+            className="autocomplete awesomplete"
```

Comment:
I specified: `container: () => containerRef.current,` when initializing awesomplete, so it uses my container. It's not going to have 2 elements with this class.

However, I nested my div inside form instead of using form as a container, because otherwise, I must overwrite styles, that awesomplete puts on elements it injected by `awesomplete > ul`. When it's nested like I did, this selector finds nothing, and I don't need to overwrite the styles. Is there a more sensible way to do that, you think?